plugins/ExamplePlugin/src/CorrelationPlot.py

```diff
@@ -0,0 +1,113 @@
+"""
+  Author:  talbpaul
+  Date  :  2021-04-02
+"""
+import os
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+from PluginBaseClasses.OutStreamPlotPlugin import PlotPlugin, InputTypes, InputData
+
+
+class Correlation(PlotPlugin):
+  # Example Plot plugin class
+  @classmethod
+  def getInputSpecification(cls):
+    """
+      Define the acceptable user inputs for this class.
+      @ In, None
+      @ Out, specs, InputData.ParameterInput,
+    """
+    specs = super().getInputSpecification()
+    specs.addSub(InputData.parameterInputFactory('bins', contentType=InputTypes.IntegerType))
+    specs.addSub(InputData.parameterInputFactory('variables', contentType=InputTypes.StringListType))
+    specs.addSub(InputData.parameterInputFactory('source', contentType=InputTypes.StringType))
+    return specs
+
+  def __init__(self):
+    """
+      Constructor.
+      @ In, None
+      @ Out, None
+    """
+    super().__init__()
+    self.printTag = 'ExamplePlugin.Correlation'
+    self._numBins = 10      # number of bins to use; np default is 10 currently
+    self._vars = None       # list of variables to plot correlations for
+    self._sourceName = None # name of source data object
+    self._source = None     # actual source data object
+
+  def handleInput(self, spec):
+    """
+      Reads in data from the input file
+      @ In, spec, InputData.ParameterInput, input information
+      @ Out, None
+    """
+    super().handleInput(spec)
+    for node in spec.subparts:
+      if node.getName() == 'bins':
+        self._numBins = node.value
+      elif node.getName() == 'variables':
+        self._vars = node.value
+      elif node.getName() == 'source':
+        self._sourceName = node.value
+    # input checking
+    if self._vars is None:
```

Comment:
Can we use InputData to check it?

Reply:
we don't have a "required" option yet for InputSpecs nodes (only for parameters/attributes)